### DISCUSSION TITLE
Builds stuck in queued state

### DISCUSSION THREAD

**Discussion by banaszeknorbert** [codemagic-ci-cd/codemagic-docs#2358] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Start new build

### Expected results

Build runs without problems. Is there a place where I can find current status of CodeMagic services?

<img width="1098" alt="Screenshot 2024-02-05 at 18 37 42" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/10154105/102c5024-d0d0-4fc6-b1cc-565b9e501692">

### Actual results

All build (for all projects in different teams) are in queued state for few hours.

**Comment by sam-parks** [comment]
I am also experiencing this

**Reply by mohamedbrahimi** [reply]
I am also having this, and it's block our work :(

**Comment by raphaelcruzeiro** [comment]
Same here. Our builds get stuck in `queued` indefinitely.

**Comment by ahmdaeyz** [comment]
We've been experiencing this for about an hour now.

**Comment by icarusdust** [comment]
Hi @banaszeknorbert @raphaelcruzeiro @ahmdaeyz, while I am checking it, please change the Xcode version in YAML to either 15.1 or 15.3.

```
environment:  
   vars:
   xcode: 15.1 or 15.3
```

The same applies to the Workflow Editor. Change its version from the dropdown. 

Note: If your Android apps are built with M1 machines, then you will need to define the Xcode version like above. It is because Xcode is pre-installed on Mac machines.

**Reply by ahmdaeyz** [reply]
Setting Xcode to 15.1 worked.

**Reply by atnpem** [reply]
is it a temporary fix?

**Reply by chrisjchrisj** [reply]
No Builds are starting, just hanging. WorkFlow Editor shows Xcode version 15.2 (grayed-out inaccessable) How can I changed that to 15.3 from WorkFlow Editor?

**Comment by dbmessina** [comment]
I was getting the same thing with Xcode set to latest. I changed it to 15.1 and a new build started immediately.

**Reply by atnpem** [reply]
For people using "WorkFlow Editor" without iOS target checked. You have to check-in build "iOS" and change the Xcode version to 15.1 in "Build" section. Then you have to disable step for iOS in "Distribution" section. Because you can't change the Xcode version without activate "iOS" build in your workflow.

**Comment by sobe-wt** [comment]
@icarusdust Any updates on this?

**Reply by helinanever** [reply]
The team is on it and we're hoping to make Xcode 15.2 machines available again as soon as possible. As a workaround, you may switch to other Xcode versions or Linux/Windows machines. We'll keep you posted.

**Reply by sobe-wt** [reply]
Gotcha. That's helpful. I'll try that now.

**Comment by helinanever** [comment, marked as answer]
The issue was caused by a massive amount of builds started by malicious users who consumed our machine resources. We hope to have now identified all such users as well as canceled their builds. However, as there are many legitimate user builds in the queue, it may still take some time for the build queue to resolve. Thank you for your understanding! See also https://status.codemagic.io/
